gateway/sensor.go

```diff
@@ -112,13 +122,18 @@ func (g *Gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		return err
 	}
 
+	return g.reconfigure(ctx, cfg)
+}
```

Comment:
I'd rather not do two reconfigures.

Reply:
why? it's much cleaner.

Reply:
We can just add the lines that you've added to the private reconfigure to the public one if it's necessary, it's probably best anyway, we still want to figure out the gateway <> sensor reconfigure flow with channel handling and rejoin requests. This module might be an always rebuild one in the end. Jury's still out.